src/ConfigLoader.php

```diff
@@ -0,0 +1,71 @@
+<?php
+namespace App;
+
+use ArrayObject;
+use RuntimeException;
+use Zend\Stdlib\ArrayUtils;
+
+class ConfigLoader
+{
+    /**
+     * @var ArrayObject
+     */
+    private $config;
+
+    private function loadConfigFromProviders(array $providers)
+    {
+        $config = [];
+        foreach ($providers as $providerClass) {
+            if (!class_exists($providerClass)) {
+                throw new RuntimeException("Cannot read config from $providerClass - class cannot be loaded.");
+            }
+            $provider = new $providerClass();
+            if (!$provider instanceof ConfigProviderInterface) {
+                throw new RuntimeException("Cannot read config from $providerClass - class does not implement ConfigProviderInterface");
+            }
+            $config = ArrayUtils::merge($config, $provider->getConfig());
+        }
+        return $config;
+    }
+
+    private function loadConfigFromFiles(array $configFiles)
+    {
+        $config = [];
+        // Load configuration from autoload path
+        foreach ($configFiles as $file) {
+            $config = ArrayUtils::merge($config, include $file);
+        }
+        return $config;
+    }
+
+    public function __construct(array $configFiles, array $providers, $cachedConfigFile = 'data/cache/app_config.php')
```

Comment:
I reused directly what was already here in `config/config.php` and I see another PR (#24) that sets it back to PHP array... once my idea is accepted I'll sort it out.